packages/shared/apis/SceneStateStorageController/utils.ts

```diff
@@ -16,6 +27,76 @@ const HUMAN_READABLE_TO_ID: Map<string, number> = new Map([
   ['Script', CLASS_ID.SMART_ITEM]
 ])
 
+export async function deployScene(payload: PublishPayload): Promise<DeploymentResult> {
+  let result: DeploymentResult
+  try {
+    // Create content client
+    const contentUrl = getFetchContentServer(store.getState())
+    const contentClient = new ContentClient(contentUrl, 'builder in-world')
+
+    // Build files
+    const entityFiles: Map<string, Buffer> = new Map()
+    for (const fileKey in payload.files) {
+      entityFiles.set(fileKey, Buffer.from(JSON.stringify(payload.files[fileKey])))
+    }
+
+    // Decode those who need it
+    for (const fileKey in payload.filesToDecode) {
+      entityFiles.set(fileKey, await blobToBuffer(payload.filesToDecode[fileKey]))
+    }
+
+    // Prepare to get the assets
+    const net = getSelectedNetwork(store.getState())
+    const baseAssetUrl = net === ETHEREUM_NETWORK.MAINNET ? BUILDER_SERVER_URL : BASE_BUILDER_SERVER_URL_ROPSTEN
+
+    // Get the assets
+    const assetsRaw: BuilderAsset[] = payload.files[CONTENT_PATH.ASSETS] as unknown as BuilderAsset[]
+    const assets = new Map<AssetId, Asset>()
+
+    for (let i = 0; i < assetsRaw.length; i++) {
+      const convertedAsset = builderAssetToLocalAsset(assetsRaw[i], baseAssetUrl)
+      assets.set(convertedAsset.id, convertedAsset)
+    }
+
+    // Generate game file
+    const gameFile: string = createGameFile(serializeSceneStateFromEntities(payload.statelessManifest.entities), assets)
+
+    entityFiles.set(CONTENT_PATH.BUNDLED_GAME_FILE, Buffer.from(gameFile))
+
+    // Build the entity
+    const { files, entityId } = await contentClient.buildEntity({
+      type: EntityType.SCENE,
+      pointers: payload.pointers,
+      files: entityFiles,
+      metadata: payload.metadata
+    })
+
+    // Sign entity id
+    const identity = getCurrentIdentity(store.getState())
+    if (!identity) {
+      throw new Error('Identity not found when trying to deploy an entity')
+    }
+    const authChain = Authenticator.signPayload(identity, entityId)
+
+    // Deploy entity
+    await contentClient.deployEntity({ files, entityId, authChain })
+
+    result = { ok: true }
+  } catch (error) {
+    result = { ok: false, error: `${error}` }
```

Comment:
Prevent state mutation, use plain-ol return statements

```suggestion
    return { ok: true }
  } catch (error) {
    return { ok: false, error: `${error}` }
```